### PR TITLE
Tests for ref_date issue

### PR DESCRIPTION
**Pull request by sambarluc**
8 tests which either open or open+print a dataset in 4 different ways, always using `ref_date`, plus `read_grid=True/False` and `read_grid=True, swap_dims=True/False`

### CODE REVIEW

File: xmitgcm/test/test_mds_store.py [hunk 1]
```diff
@@ -742,3 +743,84 @@ def test_drc_length(all_mds_datadirs):
...
+                                 geometry=expected['geometry'])
+
+def test_open_with_ref_date_without_swap(mds_datadirs_with_refdate):
+    """With ref_date, without grid."""
+    dirname, expected = mds_datadirs_with_refdate
+
+    ds = xmitgcm.open_mdsdataset(dirname, iters='all', prefix=['S'],
+                                 ref_date=expected['ref_date'], read_grid=True,
+                                 delta_t=expected['delta_t'], swap_dims=False,
+                                 geometry=expected['geometry'])
```
Comment (rabernat) [inline xmitgcm/test/test_mds_store.py]: @sambarluc sorry to be picky, but this is quite repetitive and consequently hard to parse.

Could you parametrize this into one single test?
https://docs.pytest.org/en/latest/parametrize.html

Comment (rabernat) [reply xmitgcm/test/test_mds_store.py]: See, for example `test_llc_dims`

File: xmitgcm/test/test_mds_store.py [hunk 2]
```diff
@@ -742,3 +743,84 @@ def test_drc_length(all_mds_datadirs):
...
+
+def test_print_with_ref_date_with_swap(mds_datadirs_with_refdate):
+    """With ref_date, without grid."""
+    dirname, expected = mds_datadirs_with_refdate
+
+    ds = xmitgcm.open_mdsdataset(dirname, iters='all', prefix=['S'],
+                                 ref_date=expected['ref_date'], read_grid=True,
+                                 delta_t=expected['delta_t'], swap_dims=True,
+                                 geometry=expected['geometry'])
+    print(ds)
```
Comment (rabernat) [inline xmitgcm/test/test_mds_store.py]: This is failing with `global_oce_llc90` not due to the time issue but due to the fact that you can't swap dims on llc.

File: xmitgcm/test/test_mds_store.py [hunk 3]
```diff
@@ -742,3 +743,84 @@ def test_drc_length(all_mds_datadirs):
...
+#
+def test_print_with_ref_date_without_grid(mds_datadirs_with_refdate):
+    """With ref_date, without grid."""
+    dirname, expected = mds_datadirs_with_refdate
+
+    ds = xmitgcm.open_mdsdataset(dirname, iters='all', prefix=['S'],
+                                 ref_date=expected['ref_date'], read_grid=False,
+                                 delta_t=expected['delta_t'],
+                                 geometry=expected['geometry'])
+    print(ds)
```
Comment (rabernat) [inline xmitgcm/test/test_mds_store.py]: Is there a simpler way to trigger the error than calling `print()` on the dataset? What about `ds.load()`. Or even better `ds.time.load()`. I just want to get down to the root of the issue and not mix it with other things.

**Comment by sambarluc** [comment]
It's a nightmare.

**Comment by sambarluc** [comment]
And, btw, my idea that it is linked to python 3 string formatting is clearly wrong, as it fails also on 2.7

**Comment by rabernat** [comment]
Is there a simpler way to trigger the error than calling `print()` on the dataset? What about `ds.load()`. Or even better `ds.time.load()`. I just want to get down to the root of the issue and not mix it with other things.

If we can sort that out, I will merge this and try on my own machine.

**Comment by sambarluc** [comment]
Coming with `ds.time.load()`

**Comment by rabernat** [comment]
Ok I will give it a try.

**Comment by rabernat** [comment]
So if you look carefully at the most recent travis-ci test for python 3.6, you see that the only failure is for swap_dims=True with llc geometry, which is a forbidden option. All of the actual tests did pass with python 3.6.
https://travis-ci.org/xgcm/xmitgcm/jobs/289563761
